deployment/terraform/resources/functions.tf

```diff
@@ -77,3 +77,29 @@ resource "azurerm_role_assignment" "function-app-animation-container-access" {
     azurerm_function_app.pcfuncs
   ]
 }
+
+resource "azurerm_role_assignment" "function-app-storage-table-data-contributor" {
+  scope                = azurerm_storage_account.pc.id
+  role_definition_name = "Storage Table Data Contributor"
+  principal_id         = azurerm_function_app.pcfuncs.identity[0].principal_id
+
+  depends_on = [
+    azurerm_function_app.pcfuncs
+  ]
+}
+
+data "azurerm_log_analytics_workspace" "log_analytics_workspace" {
```

Comment:
Since this is a LAW in a different subscription, let's make that super explicit in the name so that nobody alters it in the future thinking it's the test workspace. Something like `prod_log_analytics_workspace`.